### DISCUSSION TITLE
Player get stuck near the 90 degrees edges

### DISCUSSION THREAD

**Discussion by Glukozavr** [blackears/cyclopsLevelBuilder#70] [Q&A] [answered]
Hello everyone! 

Just strarted a project where I want to use cyclops. But found a very inconvenient behaviour - I have a platform and a character walk close to its edge it gets stuck. You can escape the stuck, but it obviously screws all dynamics of the game. 

Is there an explanation to this, maybe a workaround?

Thanks, will really appreciate the help, cause it's kind of critical to the game.

![stuck](https://github.com/blackears/cyclopsLevelBuilder/assets/2305194/da3f9c48-df53-41eb-8e30-94d0c1ecd7d1)

**Comment by blackears** [comment]
I think that's more just a general problem with collision and how your character controller is set up.  Maybe try adding a ramp.  You can also use a 'hover' style character where the character's collision object is only the chest area and you maintain height by raycasting to the ground and adjusting your character's position.

**Reply by Glukozavr** [reply]
A problem with what exactly with colision? When I use a simple Box3D for the platform the character moves smoothly and as expected. It only happends for Cyclops created level elements. And the script is basically simple Character3D with nothing special. So It's either the problem in Cyclops or with what Cyclops uses underneth from Godot

**Reply by blackears** [reply]
It's hard to tell form the image, nut it looks like the platform you're trying to move onto is at a higher elevation than what your character is currently on.  So that's my guess as to what's going wrong.  If it's not that, please post a project that demonstrates the problem and I can have a look at it.

**Reply by blackears** [reply]
You might also want to try using a capsule instead of a box for your character's collision object.

**Reply by Glukozavr** [reply]
I think you might have caught this wrong. I don't expect a character to step on the platform. In the GIF I am constantly keeping forward command key down.

Expected behavior: a character moves forward, reaching the platform moves along the platform

Actually behavior: a character moves forward, reaching the platform get stuck. At this point trying to move either way is unsuccessful from first tries, until eventually the characters moves away from the platform

So the problem is that I can't move along a platform built in cyclops that is half character height, as it gets stuck. I will prepare the project to show

**Reply by Glukozavr** [reply]
So, I finally got to it, @blackears 

https://github.com/Glukozavr/cyclops-showcase

I found out that this is reproduced with the 1/2 height of blocks, but not with 1 height or higher. As you might see on the video below (a longer one is also available in the repo) when I try to go forward touching the edge of a 1/2 height block - it get stuck. And when stuck it's hard to break free of that in any direction, but eventually it does. As I mentioned before it doesn't happen with regular Godot native 1/2 height box3D

https://github.com/blackears/cyclopsLevelBuilder/assets/2305194/eda34898-c6ff-4340-aefe-2dc65ea2cd41

**Comment by blackears** [comment, marked as answer]
I'm guessing this is a problem with Godot's physics.  For a block of this height, it will be hitting your capsule just under where it starts to become round which could be causing some weird feedback the physics system isn't handling properly.  I tried using a cylinder instead and the problem went away.  

With the way Cyclops is set up, each block in your scene is using a ConvexPolygonShape3D for collision.  I tried creating a second block of the same size (without Cyclops) that was just a StaticBody with BoxShape3D collision and had the same problem.

**Reply by blackears** [reply]
Same bug with BoxShape3D
[cyclops-showcase.zip](https://github.com/blackears/cyclopsLevelBuilder/files/11960811/cyclops-showcase.zip)

**Reply by Glukozavr** [reply]
Okay, perhaps when I was testing regular box3d I didn't know that height is the reason and the box was higher. I checked and it's true, the behavior is the same with regular Godot physics, it's not an issue with Cyclops. Thanks you!
That means that eiher tweak Elipse3D for the character if possible to change the angles, or change it to Cylinder3D, or simply use higher blocks.
